src/SAML2/Attribute/Attribute.php

```diff
@@ -0,0 +1,103 @@
+<?php
+
+namespace Surfnet\SamlBundle\SAML2\Attribute;
+
+use Serializable;
+
+class Attribute implements Serializable
+{
+    /**
+     * @var string
+     */
+    private $name;
+
+    /**
+     * @var array
+     */
+    private $values;
+
+    /**
+     * @var string
+     */
+    private $urnMace;
+
+    /**
+     * @var string
+     */
+    private $urnOid;
+
+    /**
+     * @param AttributeDefinition $definition
+     * @param array $values
+     */
+    public function __construct(AttributeDefinition $definition, array $values)
+    {
+        $this->name = $definition->getName();
+        $this->urnMace = $definition->getUrnMace();
+        $this->urnOid = $definition->getUrnOid();
+        $this->values = $values;
+    }
+
+    /**
+     * @return string
+     */
+    public function getName()
+    {
+        return $this->name;
+    }
+
+    /**
+     * @return string
+     */
+    public function getUrnMace()
+    {
+        return $this->urnMace;
+    }
+
+    /**
+     * @return string
+     */
+    public function getUrnOid()
+    {
+        return $this->urnOid;
+    }
+
+    /**
+     * @return string
+     */
+    public function getValues()
+    {
+        return $this->values;
+    }
+
+    /**
+     * @param Attribute $attribute
+     * @return bool
+     */
+    public function equals(Attribute $attribute)
+    {
+        return $attribute == $this;
+    }
+
+    public function serialize()
+    {
+        return serialize([
+            $this->name,
+            $this->urnMace,
+            $this->urnOid,
+            $this->values
+        ]);
+    }
+
+    public function unserialize($serialized)
+    {
+        $data = unserialize($serialized);
```

Comment:
Just a preference for explicit control over implicit behaviour. Not that it is needed here though (as Alex already pointed out in slack :wink: